src/main.rs

```diff
@@ -127,38 +132,83 @@ impl CargoCommand {
             _ => (command, false),
         };
 
+        let message_format = match Self::extract_message_format(&mut remaining_args) {
+            Some(format) => {
+                if !format.starts_with("json") {
+                    eprintln!("error: non-JSON `message-format` is not supported");
+                    std::process::exit(1);
```

Comment:
nit: `std::` isn't needed since `process` is already imported. (showed up as a lint in my editor)

Reply:
oh neat, is that lint builtin to your editor or does it come from clippy or something? I've wanted something like this but haven't seen such a lint in my editor

Reply:
I use CLion with the IntelliJ-Rust plugin. The lint was recently added:
https://github.com/intellij-rust/intellij-rust/pull/7410
https://intellij-rust.github.io/2022/04/25/changelog-169.html

Reply:
Cool, I wonder if there would be an appetite for something like that in rust-analyzer... they have some similar stuff with e.g. merging `use` 🤔